DQM/L1TMonitorClient/src/L1TStage2EMTFDEClient.cc

```diff
@@ -0,0 +1,391 @@
+    
+    EtaDif->Add(EtaD, EtaE);
+   
+    EtaRatio->Divide(EtaD, EtaDif, 2, 1); 
+  }
+
+   //Eta
+  dataHist_ = igetter.get(input_dir_data_+"/"+"emtfTrackEtaCoarse");
+  emulHist_ = igetter.get(input_dir_emul_+"/"+"emtfTrackEtaCoarse");
+
+ if (dataHist_ && emulHist_){
+    TH1F *EtaCD = dataHist_->getTH1F();
+    TH1F *EtaCE = emulHist_->getTH1F();
+
+    TH1F *EtaRatioC = emtfTrackEtaCompCoarse_->getTH1F();
+    TH1F *EtaDifC = (TH1F*) EtaRatioC->Clone();
+
+    L1TStage2EMTFDEClient::fixZero(EtaCD, EtaCE); 
+    
+    EtaDifC->Add(EtaCD, EtaCE);
+   
+    EtaRatioC->Divide(EtaCD, EtaDifC, 2, 1); 
+  }
+
+  //Phi
+  dataHist_ = igetter.get(input_dir_data_+"/"+"emtfTrackPhi");
+  emulHist_ = igetter.get(input_dir_emul_+"/"+"emtfTrackPhi");
+
+ if (dataHist_ && emulHist_){
+    TH1F *PhiD = dataHist_->getTH1F();
+    TH1F *PhiE = emulHist_->getTH1F();
+
+    TH1F *PhiRatio = emtfTrackPhiComp_->getTH1F();
+    TH1F *PhiDif = (TH1F*) PhiRatio->Clone();
+
+    L1TStage2EMTFDEClient::fixZero(PhiD, PhiE); 
+    
+    PhiDif->Add(PhiD, PhiE);
+    PhiRatio->Divide(PhiD, PhiDif, 2, 1);
+ }
+ //Phi
+  dataHist_ = igetter.get(input_dir_data_+"/"+"emtfTrackPhiCoarse");
+  emulHist_ = igetter.get(input_dir_emul_+"/"+"emtfTrackPhiCoarse");
+
+ if (dataHist_ && emulHist_){
+    TH1F *PhiCD = dataHist_->getTH1F();
+    TH1F *PhiCE = emulHist_->getTH1F();
+
+    TH1F *PhiRatioC = emtfTrackPhiCompCoarse_->getTH1F();
+    TH1F *PhiDifC = (TH1F*) PhiRatioC->Clone();
+
+    L1TStage2EMTFDEClient::fixZero(PhiCD, PhiCE); 
+    
+    PhiDifC->Add(PhiCD, PhiCE);
+    PhiRatioC->Divide(PhiCD, PhiDifC, 2, 1);
+ }
+
+
+   //Pt
+  dataHist_ = igetter.get(input_dir_data_+"/"+"emtfTrackPt");
+  emulHist_ = igetter.get(input_dir_emul_+"/"+"emtfTrackPt");
+
+ if (dataHist_ && emulHist_){
+    TH1F *PtD = dataHist_->getTH1F();
+    TH1F *PtE = emulHist_->getTH1F();
+
+    TH1F *PtRatio = emtfTrackPtComp_->getTH1F(); 
+    TH1F *PtDif = (TH1F*) PtRatio->Clone();
+
+    L1TStage2EMTFDEClient::fixZero(PtD, PtE); 
+    
+    PtDif->Add(PtD, PtE);
+    PtRatio->Divide(PtD, PtDif, 2, 1);
+ }
+
+  //PtCoarse
+  dataHist_ = igetter.get(input_dir_data_+"/"+"emtfTrackPtCoarse");
+  emulHist_ = igetter.get(input_dir_emul_+"/"+"emtfTrackPtCoarse");
+
+ if (dataHist_ && emulHist_){
+    TH1F *PtCD = dataHist_->getTH1F();
+    TH1F *PtCE = emulHist_->getTH1F();
+
+    TH1F *PtRatioC = emtfTrackPtCompCoarse_->getTH1F(); 
+    TH1F *PtDifC = (TH1F*) PtRatioC->Clone();
+
+    L1TStage2EMTFDEClient::fixZero(PtCD, PtCE); 
+    
+    PtDifC->Add(PtCD, PtCE);
+    PtRatioC->Divide(PtCD, PtDifC, 2, 1);
+ }
+   //Mode
+  dataHist_ = igetter.get(input_dir_data_+"/"+"emtfTrackMode");
+  emulHist_ = igetter.get(input_dir_emul_+"/"+"emtfTrackMode");
+
+ if (dataHist_ && emulHist_){
+    TH1F *ModeD = dataHist_->getTH1F();
+    TH1F *ModeE = emulHist_->getTH1F();
+
+    TH1F *ModeRatio = emtfTrackModeComp_->getTH1F();
+    TH1F *ModeDif = (TH1F*) ModeRatio->Clone();
```

Comment:
here and elsewhere, the Clone() needs to be deleted before going out of scope (or ownership given to something else)